libafl_bolts/src/lib.rs

```diff
@@ -945,6 +945,45 @@ pub fn format_duration_hms(duration: &time::Duration) -> String {
     format!("{}h-{}m-{}s", (secs / 60) / 60, (secs / 60) % 60, secs % 60)
 }
 
+/// Format a number with thousands separators
+#[cfg(feature = "alloc")]
+#[must_use]
+pub fn format_big_number(val: u64) -> String {
+    let short = {
+        let (num, unit) = match val {
+            0..=999 => return format!("{val}"),
+            1_000..=999_999 => (1000, "K"),
+            1_000_000..=999_999_999 => (1_000_000, "M"),
+            1_000_000_000..=999_999_999_999 => (1_000_000_000, "G"),
+            _ => (1_000_000_000_000, "T"),
+        };
+        let main = val / num;
```

Comment:
maybe check num is not 0 here

Reply:
I'm not sure it can be? The `0..=99` case above returns from the function. We just use a bare number if its too small.

Reply:
I'm not sure it can be? The `0..=99` case above returns from the function. We just use a bare number if its too small.

Reply:
yeah you are right i didn't see return before 😓